### PR TITLE
[Windows] Add safe directory to gitconfig

### PR DESCRIPTION
**Pull request by AlexanderGitin**
# Description
Due to the recent security changes in git actions\checkout task is experiencing issues https://github.com/actions/checkout/issues/760. The workaround is to set git config --global --add safe.directory "*" setting. This PR adds this setting in /etc/gitconfig as it is used globally for every OS user.
#### Related issue:
https://github.com/actions/virtual-environments-internal/issues/3617
## Check list
- [X] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/git.sh [hunk 1]
```diff
@@ -2,11 +2,12 @@
 source ~/utils/utils.sh
```
Comment (al-cheb) [inline images/macos/provision/core/git.sh:3]: Why do we have change for macOS?

File: images/win/scripts/Installers/Install-Git.ps1 [hunk 2]
```diff
@@ -4,8 +4,10 @@
 ################################################################################
 
 # Install git
-# Temporary hardcode version 2.35.1 due to the issue with the actions\checkout https://github.com/actions/checkout/issues/760
-Choco-Install -PackageName git -ArgumentList '--version', "2.35.1.2", '--installargs="/VERYSILENT /NORESTART /NOCANCEL /SP- /CLOSEAPPLICATIONS /RESTARTAPPLICATIONS /o:PathOption=CmdTools /o:BashTerminalOption=ConHost /o:EnableSymlinks=Enabled /COMPONENTS=gitlfs"'
+Choco-Install -PackageName git -ArgumentList '--installargs="/VERYSILENT /NORESTART /NOCANCEL /SP- /CLOSEAPPLICATIONS /RESTARTAPPLICATIONS /o:PathOption=CmdTools /o:BashTerminalOption=ConHost /o:EnableSymlinks=Enabled /COMPONENTS=gitlfs"'
+
+echo "[safe]
```
Comment (al-cheb) [inline images/win/scripts/Installers/Install-Git.ps1:9]: Why do we need echo?

File: images/win/scripts/Installers/Install-Git.ps1 [hunk 3]
```diff
@@ -4,8 +4,12 @@
 ################################################################################
 
 # Install git
-# Temporary hardcode version 2.35.1 due to the issue with the actions\checkout https://github.com/actions/checkout/issues/760
-Choco-Install -PackageName git -ArgumentList '--version', "2.35.1.2", '--installargs="/VERYSILENT /NORESTART /NOCANCEL /SP- /CLOSEAPPLICATIONS /RESTARTAPPLICATIONS /o:PathOption=CmdTools /o:BashTerminalOption=ConHost /o:EnableSymlinks=Enabled /COMPONENTS=gitlfs"'
+Choco-Install -PackageName git -ArgumentList '--installargs="/VERYSILENT /NORESTART /NOCANCEL /SP- /CLOSEAPPLICATIONS /RESTARTAPPLICATIONS /o:PathOption=CmdTools /o:BashTerminalOption=ConHost /o:EnableSymlinks=Enabled /COMPONENTS=gitlfs"'
+
+Add-Content -Path "C:\Program Files\Git\etc\gitconfig" -Value "
```
Comment (al-cheb) [inline images/win/scripts/Installers/Install-Git.ps1:9]: Why do not we use the same approach like we do for macOS?

`git config --system --add safe.directory "*"`

Comment (miketimofeev) [reply images/win/scripts/Installers/Install-Git.ps1:9]: @al-cheb on macOS the user remains the same after the image generation but on windows the setting will be erased as the user is different

Comment (al-cheb) [reply images/win/scripts/Installers/Install-Git.ps1:9]: @miketimofeev , agree, what's why I put `system` flag.

**Review by al-cheb — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Pull request merged**